Performance tuning documentation (#32)
KohlsTechnology/prometheus_bigquery_remote_storage_adapter: README.md

```

## Performance Tuning

You will need to tune the storage adapter based on your needs. You have several levers available...

> ### Requests & limits

Reviewer: ```suggestion
### Requests & Limits
```

When running on a container platform (like Kubernetes), it's important to configure the CPU / memory requests and limits properly. You should be able to get away with just a couple hundred megabytes of RAM (make sure request == limit), but the CPU needs will heavily depend on your environment. Set the CPU requests to the minimum you need to achieve the required performance. We recommend setting the limit higher (keep in mind that anything above the request is not guaranteed). Keep an eye on CPU throttling to help tweak your settings.

> ### Limit metrics stored long-term

Reviewer: ```suggestion
### Limit Metrics Stored Long-Term
```

> The amount of data you send to BigQuery can be another big constraint. It is easy to overwhelm the BigQuery streaming engine by throwing millions of records at it. You might run into API quota issues or simply have data gaps. We highly recommend not to go crazy when it comes to scrape intervals and also being a lot more selective on what gets actually stored long-term. Depending on your needs, it might make sense to calculate aggregated metrics and store only those long-term.

Reviewer: ```suggestion
The amount of data you send to BigQuery can be another big constraint. It is easy to overwhelm the BigQuery streaming engine by throwing millions of records at it. You might run into API quota issues or simply have data gaps. We highly recommend not to go crazy when it comes to scrape intervals (<30s) and be very selective on what gets stored long-term. Depending on your needs, it might make sense to calculate and store only aggregated metrics long-term.
```

Refer to the Prometheus documentation for [remote_write](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write) and [relabel_config](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config) on how to implement this.

> ### Prometheus remote_write queue_config

Reviewer: ```suggestion
### Prometheus Remote Storage (remote_write & queue_config)
```
